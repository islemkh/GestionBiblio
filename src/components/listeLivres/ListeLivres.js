import React, { useState, useCallback, useEffect } from "react"
import Livre from "../livres/Livre"

 
function ListeLivres({livres}){

  return (
   <center> <div className="adherentsTab-list">
      <h1>Listes des livres</h1>
     <div className="listeStyle"  >
        {livres.map((livre, index)=> (
          <Livre
            key={index}
              id={livre.id}
              titre={livre.titre}
              auteur={livre.auteur}
              edition={livre.edition}  
            />
        ))}
      </div>
    </div>
    </center>
  )

}
export default  ListeLivres