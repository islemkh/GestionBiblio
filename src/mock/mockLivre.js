export default [
    {
        id:"1",
        titre: "Le Grand Meaulnes",
        auteur: "Victor Hugo",
        edition: 2015,
        NbExemplaires: 20,
        etat : " non archivé"
      }, 
      {
        id:"2",
        titre: "Le Rouge et le Noir",
        auteur: "Victor Del Litto ",
        edition: 1997,
        NbExemplaires: 5,
        etat : "archivé"
      }, 
      {
        id:"3",
        titre: "Le Petit Prince",
        auteur: "Antoine de Saint-Exupéry",
        edition: 2014,
        NbExemplaires: 10,
        etat : "archivé"
      }, 
      {
        id:"4",
        titre: "Lettres de mon moulin",
        auteur: "Alphonse Daudet",
        edition: 2005,
        NbExemplaires: 12,
        etat : "archivé"
      }, 
      {
        id:"5",
        titre: "La Mère",
        auteur: "Pearl Buck",
        edition: 1971,
        NbExemplaires: 30,
        etat : "non archivé"
      },
      {
        id:"6",
        titre: "Les Fourmis",
        auteur: "Bernard Werber",
        edition: 2002,
        NbExemplaires: 8,
        etat : "non archivé"
      },
       {
        id:"7",
        titre: "Si c'est un homme",
        auteur: "Primo Levi",
        edition: 2007,
        NbExemplaires: 6,
        etat : "non archivé"

      }
    ]