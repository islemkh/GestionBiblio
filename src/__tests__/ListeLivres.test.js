import React from 'react';
import { render } from '@testing-library/react';
import ListeLivres from '../components/listeLivres/ListeLivres';
import mockLivre from '../mock/mockLivre'

describe("test list livres",()=>{
    test("",()=>{
      /*  // console.log(mockLivre)
       const LivresComponent = render(
        <ListeLivres livres={mockLivre} />
)
        expect(LivresComponent).toMatchSnapshot() */
    })
})