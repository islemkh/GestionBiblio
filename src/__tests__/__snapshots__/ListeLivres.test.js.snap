// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test list livres  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <center>
         
        <div
          class="adherentsTab-list"
        >
          <h1>
            Listes des livres
          </h1>
          <div
            class="listeStyle"
          >
            <td
              class="MuiTableCell-root"
            >
              Le Grand Meaulnes
            </td>
            <td
              class="MuiTableCell-root"
            >
              Victor Hugo
            </td>
            <td
              class="MuiTableCell-root"
            >
              <div>
                <button
                  class="emprunter"
                >
                  Emprunter
                </button>
                <button
                  class="details"
                >
                  Details
                </button>
              </div>
            </td>
            <td
              class="MuiTableCell-root"
            >
              Le Rouge et le Noir
            </td>
            <td
              class="MuiTableCell-root"
            >
              Victor Del Litto 
            </td>
            <td
              class="MuiTableCell-root"
            >
              <div>
                <button
                  class="emprunter"
                >
                  Emprunter
                </button>
                <button
                  class="details"
                >
                  Details
                </button>
              </div>
            </td>
            <td
              class="MuiTableCell-root"
            >
              Le Petit Prince
            </td>
            <td
              class="MuiTableCell-root"
            >
              Antoine de Saint-Exupéry
            </td>
            <td
              class="MuiTableCell-root"
            >
              <div>
                <button
                  class="emprunter"
                >
                  Emprunter
                </button>
                <button
                  class="details"
                >
                  Details
                </button>
              </div>
            </td>
            <td
              class="MuiTableCell-root"
            >
              Lettres de mon moulin
            </td>
            <td
              class="MuiTableCell-root"
            >
              Alphonse Daudet
            </td>
            <td
              class="MuiTableCell-root"
            >
              <div>
                <button
                  class="emprunter"
                >
                  Emprunter
                </button>
                <button
                  class="details"
                >
                  Details
                </button>
              </div>
            </td>
            <td
              class="MuiTableCell-root"
            >
              La Mère
            </td>
            <td
              class="MuiTableCell-root"
            >
              Pearl Buck
            </td>
            <td
              class="MuiTableCell-root"
            >
              <div>
                <button
                  class="emprunter"
                >
                  Emprunter
                </button>
                <button
                  class="details"
                >
                  Details
                </button>
              </div>
            </td>
            <td
              class="MuiTableCell-root"
            >
              Les Fourmis
            </td>
            <td
              class="MuiTableCell-root"
            >
              Bernard Werber
            </td>
            <td
              class="MuiTableCell-root"
            >
              <div>
                <button
                  class="emprunter"
                >
                  Emprunter
                </button>
                <button
                  class="details"
                >
                  Details
                </button>
              </div>
            </td>
            <td
              class="MuiTableCell-root"
            >
              Si c'est un homme
            </td>
            <td
              class="MuiTableCell-root"
            >
              Primo Levi
            </td>
            <td
              class="MuiTableCell-root"
            >
              <div>
                <button
                  class="emprunter"
                >
                  Emprunter
                </button>
                <button
                  class="details"
                >
                  Details
                </button>
              </div>
            </td>
          </div>
        </div>
      </center>
    </div>
  </body>,
  "container": <div>
    <center>
       
      <div
        class="adherentsTab-list"
      >
        <h1>
          Listes des livres
        </h1>
        <div
          class="listeStyle"
        >
          <td
            class="MuiTableCell-root"
          >
            Le Grand Meaulnes
          </td>
          <td
            class="MuiTableCell-root"
          >
            Victor Hugo
          </td>
          <td
            class="MuiTableCell-root"
          >
            <div>
              <button
                class="emprunter"
              >
                Emprunter
              </button>
              <button
                class="details"
              >
                Details
              </button>
            </div>
          </td>
          <td
            class="MuiTableCell-root"
          >
            Le Rouge et le Noir
          </td>
          <td
            class="MuiTableCell-root"
          >
            Victor Del Litto 
          </td>
          <td
            class="MuiTableCell-root"
          >
            <div>
              <button
                class="emprunter"
              >
                Emprunter
              </button>
              <button
                class="details"
              >
                Details
              </button>
            </div>
          </td>
          <td
            class="MuiTableCell-root"
          >
            Le Petit Prince
          </td>
          <td
            class="MuiTableCell-root"
          >
            Antoine de Saint-Exupéry
          </td>
          <td
            class="MuiTableCell-root"
          >
            <div>
              <button
                class="emprunter"
              >
                Emprunter
              </button>
              <button
                class="details"
              >
                Details
              </button>
            </div>
          </td>
          <td
            class="MuiTableCell-root"
          >
            Lettres de mon moulin
          </td>
          <td
            class="MuiTableCell-root"
          >
            Alphonse Daudet
          </td>
          <td
            class="MuiTableCell-root"
          >
            <div>
              <button
                class="emprunter"
              >
                Emprunter
              </button>
              <button
                class="details"
              >
                Details
              </button>
            </div>
          </td>
          <td
            class="MuiTableCell-root"
          >
            La Mère
          </td>
          <td
            class="MuiTableCell-root"
          >
            Pearl Buck
          </td>
          <td
            class="MuiTableCell-root"
          >
            <div>
              <button
                class="emprunter"
              >
                Emprunter
              </button>
              <button
                class="details"
              >
                Details
              </button>
            </div>
          </td>
          <td
            class="MuiTableCell-root"
          >
            Les Fourmis
          </td>
          <td
            class="MuiTableCell-root"
          >
            Bernard Werber
          </td>
          <td
            class="MuiTableCell-root"
          >
            <div>
              <button
                class="emprunter"
              >
                Emprunter
              </button>
              <button
                class="details"
              >
                Details
              </button>
            </div>
          </td>
          <td
            class="MuiTableCell-root"
          >
            Si c'est un homme
          </td>
          <td
            class="MuiTableCell-root"
          >
            Primo Levi
          </td>
          <td
            class="MuiTableCell-root"
          >
            <div>
              <button
                class="emprunter"
              >
                Emprunter
              </button>
              <button
                class="details"
              >
                Details
              </button>
            </div>
          </td>
        </div>
      </div>
    </center>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
